### PR TITLE
Resolve avro refs for wrapped unions

### LINKED ISSUE #37: avro-ts - types for some wrapped unions are incorrect
I think this schema

```
{
  type: 'record',
  name: 'Event',
  namespace: 'com.example.avro',
  fields: [
    {
      name: 'field',
      type: [
        {
          type: 'record',
          name: 'FieldTypeA',
          fields: [
            { name: 'id', type: 'string' },
            {
              name: 'nested',
              type: {
                name: 'NestedFieldType',
                type: 'record',
                fields: [
                  {name: 'id', type: 'string'}
                ]
              }
            }
          ],
        },
        {
          type: 'record',
          name: 'FieldTypeB',
          fields: [
            { name: 'id', type: 'string' },
          ],
        },
        'Field.NestedFieldType'      // <-- Nested type referred to by name
      ],
    }
  ],
}
```

should result in this Event:

```
export interface Event {
        field: {
            "com.example.avro.FieldTypeA": ComExampleAvro.FieldTypeA;
            "com.example.avro.FieldTypeB"?: never;
            "com.example.avro.NestedFieldType"?: never;
        } | {
            "com.example.avro.FieldTypeA"?: never;
            "com.example.avro.FieldTypeB": ComExampleAvro.FieldTypeB;
            "com.example.avro.NestedFieldType"?: never;
        } | {
            "com.example.avro.FieldTypeA"?: never;
            "com.example.avro.FieldTypeB"?: never;
            "com.example.avro.NestedFieldType": ComExampleAvro.NestedFieldType;
        };
 }
```

whereas instead, it results in this:
```
export interface Event {
        field: ComExampleAvro.FieldTypeA | ComExampleAvro.FieldTypeB | Field.NestedFieldType;
}
```

If we copy the nested type instead of referring to it by its type name, as per the schema below, we get the expected types
```
{
  type: 'record',
  name: 'Event',
  namespace: 'com.example.avro',
  fields: [
    {
      name: 'field',
      type: [
        {
          type: 'record',
          name: 'FieldTypeA',
          fields: [
            { name: 'id', type: 'string' },
            {
              name: 'nested',
              type: {
                name: 'NestedFieldType',
                type: 'record',
                fields: [
                  {name: 'id', type: 'string'}
                ]
              }
            }
          ],
        },
        {
          type: 'record',
          name: 'FieldTypeB',
          fields: [
            { name: 'id', type: 'string' },
          ],
        },
        {                                                 // <-- copied
          name: 'NestedFieldType',
          type: 'record',
          fields: [
            {name: 'id', type: 'string'}
          ]
        }
      ],
    }
  ],
}
```

This is an issue when generating types for the `comms_communication_update_v1-value` schema which uses a wrapped union for the `deliverTo` property

### PR DESCRIPTION
**Pull request by ivank**
Should fix #37

### CODE REVIEW

File: packages/avro-ts/src/convert.ts [hunk 1]
```diff
@@ -29,8 +27,30 @@ export const nameParts = (fullName: string): [string] | [string, string] => {
...
+          ...all,
+          namespace: type.namespace ?? all.namespace,
+          refs: { ...all.refs, [fullName(all, type)]: type },
+        }),
+      context,
+    );
+  } else {
+    return context;
+  }
+};
```
Comment (samjacobclift) [inline packages/avro-ts/src/convert.ts:50]: I was thinking if there is a nicer way to write this with a switch statement, but I couldn't figure out a way that wouldn't just end up with more code!

File: packages/avro-ts/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@ovotech/avro-ts",
   "description": "Convert avro schemas into typescript interfaces",
-  "version": "4.2.2",
+  "version": "4.2.3",
```
[the remark this answers is not in the document]
Comment (ivank) [reply packages/avro-ts/package.json:4]: Well maybe, but isn't it a bug more than anything right?

[the remark this answers is not in the document]
Comment (samjacobclift) [reply packages/avro-ts/package.json:4]: :shipit:

File: packages/avro-ts/test/__snapshots__/integration.spec.ts.snap [hunk 3]
```diff
@@ -623,6 +623,45 @@ export namespace ComAvroExample {
...
+    export const FieldTypeBName = \\"com.example.avro.FieldTypeB\\";
+    export interface FieldTypeB {
+        id: string;
+    }
+    export const EventName = \\"com.example.avro.Event\\";
+    export interface Event {
+        field: {
+            \\"com.example.avro.FieldTypeA\\": ComExampleAvro.FieldTypeA;
+            \\"com.example.avro.FieldTypeB\\"?: never;
+            \\"com.example.avro.undefined\\"?: never;
```
Comment (tarlingovo) [inline packages/avro-ts/test/__snapshots__/integration.spec.ts.snap:650]: Hi @ivank . I don't think these `undefined`s are correct. I'll also take a look and see if I can fix that

**Pull request merged**